<?xml version="1.0" encoding="UTF-8"?>
<testng-results retried="6" ignored="0" total="8" passed="0" failed="2" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-08-12T01:04:30 GMT-04:00" name="Suite_dsAlgo" finished-at="2025-08-12T01:05:00 GMT-04:00" duration-ms="30229">
    <groups>
    </groups>
    <test started-at="2025-08-12T01:04:30 GMT-04:00" name="Test_All_Modules_chrome" finished-at="2025-08-12T01:05:00 GMT-04:00" duration-ms="30229">
      <class name="testCases.Stack">
        <test-method signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:30 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:05:00 GMT-04:00" duration-ms="30031" status="FAIL">
          <exception class="org.openqa.selenium.UnhandledAlertException">
            <message>
              <![CDATA[unexpected alert open: {Alert text : NameError: name 'DsAlgoNew' is not defined on line 1}
  (Session info: chrome=138.0.7204.185): NameError: name 'DsAlgoNew' is not defined on line 1
Build info: version: '4.33.0', revision: '2c6aaad03a'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.7'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [507f59f2cf298a608dbf856b7ab8a11b, screenshot {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 138.0.7204.185, chrome: {chromedriverVersion: 138.0.7204.183 (e90faf484dd..., userDataDir: C:\Users\HP\AppData\Local\T...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:49271}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:49271/devtoo..., se:cdpVersion: 138.0.7204.185, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 507f59f2cf298a608dbf856b7ab8a11b]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.UnhandledAlertException: unexpected alert open: {Alert text : NameError: name 'DsAlgoNew' is not defined on line 1}
  (Session info: chrome=138.0.7204.185): NameError: name 'DsAlgoNew' is not defined on line 1
Build info: version: '4.33.0', revision: '2c6aaad03a'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.7'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [507f59f2cf298a608dbf856b7ab8a11b, screenshot {}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 138.0.7204.185, chrome: {chromedriverVersion: 138.0.7204.183 (e90faf484dd..., userDataDir: C:\Users\HP\AppData\Local\T...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:49271}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), se:cdp: ws://localhost:49271/devtoo..., se:cdpVersion: 138.0.7204.185, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 507f59f2cf298a608dbf856b7ab8a11b
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:136)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:50)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:215)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:544)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:617)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:621)
at org.openqa.selenium.remote.RemoteWebDriver.getScreenshotAs(RemoteWebDriver.java:333)
at testCases.BaseTest.failed_screenshot(BaseTest.java:60)
at testCases.CustomListener.onTestFailure(CustomListener.java:18)
at org.testng.internal.TestListenerHelper.runTestListeners(TestListenerHelper.java:99)
at org.testng.internal.invokers.TestInvoker.runTestResultListener(TestInvoker.java:277)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:764)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.UnhandledAlertException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:30 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:32 GMT-04:00" duration-ms="2784" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:32 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:33 GMT-04:00" duration-ms="103" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:33 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:34 GMT-04:00" duration-ms="1122" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[print("Hello World")]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:34 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:34 GMT-04:00" duration-ms="193" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:34 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:36 GMT-04:00" duration-ms="1634" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:36 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:36 GMT-04:00" duration-ms="85" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:36 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:37 GMT-04:00" duration-ms="1630" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[print("Hello World")]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:37 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:38 GMT-04:00" duration-ms="220" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:38 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:40 GMT-04:00" duration-ms="2165" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:40 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:40 GMT-04:00" duration-ms="73" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:40 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:41 GMT-04:00" duration-ms="1183" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[print("Hello World")]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:41 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:41 GMT-04:00" duration-ms="171" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:41 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:43 GMT-04:00" duration-ms="2208" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:43 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:43 GMT-04:00" duration-ms="68" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:43 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:45 GMT-04:00" duration-ms="1269" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[print("Hello World")]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:45 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:45 GMT-04:00" duration-ms="204" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:45 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:47 GMT-04:00" duration-ms="2120" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:47 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:47 GMT-04:00" duration-ms="49" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:47 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:48 GMT-04:00" duration-ms="1115" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[DsAlgoNew]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:48 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:49 GMT-04:00" duration-ms="178" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:49 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:51 GMT-04:00" duration-ms="2226" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:51 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:51 GMT-04:00" duration-ms="72" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:51 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:52 GMT-04:00" duration-ms="1224" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[DsAlgoNew]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:52 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:52 GMT-04:00" duration-ms="196" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:52 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:55 GMT-04:00" duration-ms="2316" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:55 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:55 GMT-04:00" duration-ms="68" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
        <test-method retried="true" signature="StackoperationsTryingValidAndInvalidCode(java.lang.String)[pri:4, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:55 GMT-04:00" name="StackoperationsTryingValidAndInvalidCode" data-provider="pythonCodeValidandInvalid" finished-at="2025-08-12T01:04:56 GMT-04:00" duration-ms="1199" status="SKIP">
          <params>
            <param index="0">
              <value>
                <![CDATA[DsAlgoNew]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.HashMap.get(Object)" is null
at utils.ExcelReader.getData(ExcelReader.java:86)
at pageFactory.BasePage.inValidCode(BasePage.java:121)
at testCases.Stack.StackoperationsTryingValidAndInvalidCode(Stack.java:84)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:475)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- StackoperationsTryingValidAndInvalidCode -->
        <test-method is-config="true" signature="teardownDriver()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:56 GMT-04:00" name="teardownDriver" finished-at="2025-08-12T01:04:56 GMT-04:00" duration-ms="208" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardownDriver -->
        <test-method is-config="true" signature="open_website(java.lang.String)[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:56 GMT-04:00" name="open_website" finished-at="2025-08-12T01:04:58 GMT-04:00" duration-ms="2397" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- open_website -->
        <test-method is-config="true" signature="stack_page()[pri:0, instance:testCases.Stack@53ed09e8]" started-at="2025-08-12T01:04:58 GMT-04:00" name="stack_page" finished-at="2025-08-12T01:04:59 GMT-04:00" duration-ms="58" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stack_page -->
      </class> <!-- testCases.Stack -->
    </test> <!-- Test_All_Modules_chrome -->
  </suite> <!-- Suite_dsAlgo -->
</testng-results>
